pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadRestletResource.java

```diff
@@ -352,10 +352,11 @@ private String getZkDownloadURIForSegmentUpload(FileUploadPathProvider provider,
     if (baseDataDirURI.getScheme().equalsIgnoreCase(CommonConstants.Segment.LOCAL_SEGMENT_SCHEME)) {
       return URIUtils.constructDownloadUrl(provider.getVip(), rawTableName, segmentName);
     } else {
-      // Receiving .tar.gz segment upload for pluggable storage
-      LOGGER.info("Using configured data dir {} for segment {} of table {}", _controllerConf.getDataDir(), segmentName,
-          rawTableName);
-      return URIUtils.constructDownloadUrl(baseDataDirURI.toString(), rawTableName, segmentName);
+      // Receiving .tar.gz segment upload for pluggable storage. Download URI is the same as final segment location.
+      String downloadUri = URIUtils.getUri(baseDataDirURI.toString(), rawTableName, URIUtils.encode(segmentName)).toString();
+      LOGGER.info("Using configured data dir {} for segment {} of table {}, Download URI: {}", _controllerConf.getDataDir(),
```

Comment:
Please revise the log info, the data dir in log is not in sync with the behavior.
Suggest changing it to "Using download URI: {} for segment: {} of table: {}"

Reply:
:+1: